### ISSUE TITLE
builds from docker images do not keep labels

### ISSUE THREAD

**Issue by EvanTheB** [apptainer/singularity#3984]
## Version of Singularity:

singularity --version
singularity version 3.2.0-rc2

### Expected behavior

images built from docker have the docker image labels in the singularity image labels.

### Actual behavior

It doesn't happen, silently.

### Steps to reproduce behavior

```
echo 'FROM debian
LABEL testl=labelv' | docker build -t debtest -

singularity build test docker-daemon://debtest:latest

singularity inspect test
```

I tried to investigate myself, but got stumped at this obscure line:

https://github.com/sylabs/singularity/blob/master/internal/pkg/build/sources/conveyorPacker_oci.go#L25

Which doesn't seem to refer to a real git repo?

**Comment by jscook2345** [comment]
@EvanTheB 

The link refers to: https://github.com/containers/image/tree/master/docker/daemon

GitHub shows things based on branches, so you need to include the branch when viewing it in your browser.

Let me look into the label part quickly and get back to you!

**Comment by jscook2345** [comment]
It appears we do not do this currently. I've added this as an enhancement that we may fix in the future. Thanks for the report @EvanTheB !

**Comment by EvanTheB** [comment]
Thanks @jscook2345, I understand that the docker-daemon is not a 'real' builder. But it seems extremely useful to me, and I see a few questions around asking how to build from local docker images without having to make a local registry. Hopefully it can be promoted to a real builder someday. Maybe the normal docker builder could be a 2 stop `docker pull` and then `docker-daemon build`. That way the behaviour is the same between the 2 processes, and functionality does not have to be duplicated.

For now here is my workaround for others:

```
echo 'Bootstrap: docker-daemon
From: '"$name:$label" > tmp.singularity.def

echo '%labels' >> tmp.singularity.def
docker inspect "$docker_label_id" \
| jq -r '.[].ContainerConfig.Labels
| to_entries
| map([.key, .value] | join(" "))
| join("\n")' >> tmp.singularity.def

```

**Comment by jscook2345** [comment]
@EvanTheB 

We do consider it a real builder (though perhaps a little undocumented), but we do not explicitly import the labels as you've demonstrated. I'm not sure what the history is on that so I can't speak to why we do not.

@ikaneshiro ?

**Comment by carterpeel** [comment]
Hello,

This is a templated response that is being sent out to all open issues.  We are working hard on 'rebuilding' the Singularity community, and a major task on the agenda is finding out what issues are still outstanding.


**Please consider the following:**

1. Is this issue a duplicate, or has it been fixed/implemented since being added?
2. Is the issue still relevant to the current state of Singularity's functionality?
3. Would you like to continue discussing this issue or feature request?


Thanks,
Carter

**Comment by rdenham** [comment]
I'd like to point out that this is not restricted to just docker-daemon, but docker in general. If you build a docker image with a label, and push it to a registry you get the same result - the labels don't carry over.

I had a quick look for an image with a label on dockerhub (turns out labels aren't used that often), but here is an example:

```
singularity pull docker://couchbase
singularity inspect -j couchbase_latest.sif 
```

Produces
```
WARNING: No SIF metadata partition, searching in container...
{
        "data": {
                "attributes": {
                        "labels": {
                                "org.label-schema.build-date": "Friday_21_May_2021_9:15:42_AEST",
                                "org.label-schema.schema-version": "1.0",
                                "org.label-schema.usage.singularity.deffile.bootstrap": "docker",
                                "org.label-schema.usage.singularity.deffile.from": "couchbase",
                                "org.label-schema.usage.singularity.version": "3.5.3"
                        }
                }
        },
        "type": "container"
}
```

But checking the [Dockerfile](https://github.com/couchbase/docker/blob/master/community/couchbase-server/6.6.0/Dockerfile) shows that it has the label:

```
LABEL maintainer="docker@couchbase.com"
```

Further, 

```
singularity sif dump 2 couchbase_latest.sif 
```

gives 

```
{"ExposedPorts":{"11207/tcp":{},"11210/tcp":{},"11211/tcp":{},"18091/tcp":{},"18092/tcp":{},"18093/tcp":{},"18094/tcp":{},"18095/tcp":{},"18096/tcp":{},"8091/tcp":{},"8092/tcp":{},"8093/tcp":{},"8094/tcp":{},"8095/tcp":{},"8096/tcp":{}},"Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/couchbase/bin:/opt/couchbase/bin/tools:/opt/couchbase/bin/install"],"Entrypoint":["/entrypoint.sh"],"Cmd":["couchbase-server"],"Volumes":{"/opt/couchbase/var":{}},"Labels":{"maintainer":"docker@couchbase.com"}}
```
(with the label included).

I'm using singularity version 3.5.3, but have tested it on more recent versions with the same result.

I'm not sure if this is a bug or an enhancement. This did work in singularity 2.x, and early documentation suggest that they are carried over: ["Labels from Docker bootstraps will be carried forward here"](https://singularity-docs.readthedocs.io/en/latest/#labels).

This would be useful to us as a way of tracking key metadata from a docker image into singularity, so would be valuable to us if this behaviour could be reinstated.

**Comment by pedroalvesbatista** [comment]
@EvanTheB Still getting this problem ? Have this been solved already ? If yes, what work around have you followed and applied ?

We're looking into the issue carefully, soon will bring to community and discuss ways to better solve as well address this. Thankyou for keeping the interest in the subject.

**Comment by psavery** [comment]
This issue was fixed in the sylabs fork in sylabs/singularity#152. If it is not fixed here, perhaps similar changes can be made to fix it?

**Comment by DrDaveD** [comment]
We do intend to import most of the merged PRs from singularityCE so it will probably get in.

**Comment by psavery** [comment]
@DrDaveD I saw that sylabs/singularity#152 was added in #6105, and then removed in #6119. Is there still a plan to put it back in soon? I see it isn't in #6143 either.

**Comment by psavery** [comment]
Oh, maybe this is going to wait until the next feature release (`3.9.0`) rather than on a patch release?

**Comment by DrDaveD** [comment]
Yes, it will get put back in on the master branch for 3.9.0.

**Comment by DrDaveD** [comment]
This was fixed in #6188
